pandas/_libs/tslibs/conversion.pyx

```diff
@@ -605,13 +605,20 @@ cdef inline check_overflows(_TSObject obj):
     OutOfBoundsDatetime
     """
     # GH#12677
+    fmt = (f'{obj.dts.year}-{obj.dts.month:02d}-{obj.dts.day:02d} '
+           f'{obj.dts.hour:02d}:{obj.dts.min:02d}:{obj.dts.sec:02d}')
```

Comment:
can you delay evaluating this until we know we're raising?

also pls use double-quotes

Reply:
you can use a ```lambda x: ...``` here and use `.format(obj)`

Reply:
i think we can milk a tiny bit more performance by avoiding defining the lambda

Reply:
well, you end up having to actually either define everything twice (gross) or use a function call, which is precisely what i am suggesting

Reply:
> define everything twice (gross) or use a function call

Even if you're not evaluating the lambda, defining it at runtime is overhead we can avoid.

i expect the performance difference is tiny, but fstrings showed up in a perf regression @TomAugspurger fixed the other day.

Reply:
I just went the defining twice route for now, but I agree it kinda feels gross. Let me know if I should change it.